WooCommerce/WooCommerceTests/POS/Models/PointOfSaleAggregateModelTests.swift

```diff
@@ -429,6 +436,150 @@ struct PointOfSaleAggregateModelTests {
         }
     }
 
+    struct OrderTests {
+        private let cardPresentPaymentService = MockCardPresentPaymentService()
+        private let itemProvider = MockPOSItemProvider()
+        private let orderService = MockPOSOrderService()
+        private let sut: PointOfSaleAggregateModel
+
+        init() {
+            orderService.orderToReturn = Order.fake()
+
+            sut = PointOfSaleAggregateModel(
+                itemProvider: itemProvider,
+                cardPresentPaymentService: cardPresentPaymentService,
+                orderService: orderService)
+
+            sut.addToCart(makeItem())
+        }
+
+        @Test func startNewCart_sets_orderState_to_idle() async throws {
+            // Given
+            await sut.checkOut()
+            try #require(sut.orderState == .loaded(.init(
+                cartTotal: "$0.00",
+                orderTotal: "",
+                taxTotal: "")))
+
+            // When
+            sut.startNewCart()
+
+            // Then
+            #expect(sut.orderState == .idle)
+        }
+
+        @Test func checkOut_when_reader_connects_collectPayment_called() async throws {
+            // Given
+            cardPresentPaymentService.connectedReader = nil
+            await sut.checkOut()
+            cardPresentPaymentService.collectPaymentWasCalled = false
+
+            // When
+            // `await confirmation` callback only waits until this completes, not until some timeout.
+            // Since this is synchonous but triggers async combine behaviour, we can't use that approach.
+            cardPresentPaymentService.connectedReader = .init(name: "Test reader", batteryLevel: 0.7)
+
+            // Then
+            let timeout = ContinuousClock.now + .seconds(1)
+
+            while cardPresentPaymentService.collectPaymentWasCalled != true {
+                try! await Task.sleep(for: .milliseconds(1))
```

Comment:
We don't need to force-unwrap here:

```suggestion
                try await Task.sleep(for: .milliseconds(1))
```
Question: Is there any risk of infinite looping here by combining `ContinuousClock` + `while != true` ? 🤔

Reply:
No – the continuous clock use is outside the loop, so it's setting the time bound for the loop.

`#require` will both fail the test and stop further execution of it, so we don't need to specifically break the loop. If we used `#expect` it would loop forever if collectPayment was never called, or loop until it was called but fail anyway if it happened after the timeout.